File: android/app/src/main/java/app/organicmaps/sdk/bookmarks/data/TrackStatistics.java
package app.organicmaps.sdk.bookmarks.data;

import androidx.annotation.Keep;

// Used by JNI
@Keep
public class TrackStatistics
{
  private final double m_length;
  private final double m_duration;
  private final double m_ascent;
  private final double m_descent;
  private final int m_minElevation;
  private final int m_maxElevation;

  @Keep
  public TrackStatistics(double length, double duration, double ascent, double descent, int minElevation,
                         int maxElevation)
  {
    m_length = length;
    m_duration = duration;
    m_ascent = ascent;
    m_descent = descent;
    m_minElevation = minElevation;
    m_maxElevation = maxElevation;
  }

  public double getLength()
  {
    return m_length;
  }

  public double getDuration()
  {
    return m_duration;
  }

  public double getAscent()
  {
    return m_ascent;
  }

  public double getDescent()
  {
    return m_descent;
  }

  public int getMinElevation()
  {
    return m_minElevation;
  }

  public int getMaxElevation()
  {
    return m_maxElevation;
  }
}

File: android/sdk/src/main/cpp/app/organicmaps/sdk/bookmarks/data/TrackStatistics.cpp
#include "TrackStatistics.hpp"

#include "app/organicmaps/sdk/core/jni_helper.hpp"

jobject ToJavaTrackStatistics(JNIEnv * env, TrackStatistics const & trackStats)
{
  static jclass clazz = jni::GetGlobalClassRef(env, "app/organicmaps/sdk/bookmarks/data/TrackStatistics");
  // clang-format off
  static jmethodID const cId = jni::GetConstructorID(env, clazz,
    "("
    "D"  // m_length
    "D"  // m_duration
    "D"  // m_ascent
    "D"  // m_descent
    "I"  // m_minElevation
    "I"  // m_maxElevation
    ")V"
  );
  return env->NewObject(clazz, cId,
    trackStats.m_length,
    trackStats.m_duration,
    trackStats.m_ascent,
    trackStats.m_descent,
    static_cast<jint>(trackStats.m_minElevation),
    static_cast<jint>(trackStats.m_maxElevation)
  );
  // clang-format on
}
